In [195]:
import pandas as pd

# Carregar os dados da planilha Excel
df = pd.read_excel('Questionário Artes Gestacional (respostas).xlsx')

df.to_csv('base.csv', sep=";")

In [196]:
# Funções pra contar resposta única e múltiplas respostas

def contar_resposta_unica(coluna, df):
    contagem = df[coluna].value_counts()
    porcentagens = (contagem / contagem.sum()) * 100
    porcentagens = porcentagens.round(2)
    df_resultado = pd.DataFrame({coluna: contagem.index, 'Contagem': contagem.values, 'Porcentagem': porcentagens.values})
    print(df_resultado)
    print("\n")
    return df_resultado

def contar_multiplas_respostas(coluna, df):
    # Separar as respostas em colunas diferentes
    respostas_sep = df[coluna].str.split(', ', expand=True)
    
    # Contar as ocorrências de cada comportamento em todas as respostas
    ocorrencias = respostas_sep.apply(pd.Series.value_counts).fillna(0)
    
    # Converter os dados para inteiros
    ocorrencias = ocorrencias.astype(int)
    
    # Somar as contagens de cada comportamento em todas as colunas
    ocorrencias_total = ocorrencias.sum(axis=1)
    
    # Calcular as porcentagens
    total_respostas = len(df)
    porcentagens_total = (ocorrencias_total / total_respostas) * 100
    porcentagens_total = porcentagens_total.round(2)
    
    # Criar DataFrame com contagem e porcentagem
    resultado_df = pd.DataFrame({coluna: ocorrencias_total.index, 'Contagem': ocorrencias_total.values, 'Porcentagem': porcentagens_total.values})
    print(resultado_df)
    print("\n")
    return resultado_df


In [197]:
# Função para classificar a idade gestacional em trimestres
def classificar_trimestre(idade):
    try:
        semanas = int(idade.split('s')[0])
        if semanas >= 1 and semanas <= 13:
            return 'Primeiro Trimestre'
        elif semanas >= 14 and semanas <= 26:
            return 'Segundo Trimestre'
        elif semanas >= 27 and semanas <= 41:
            return 'Terceiro Trimestre'
        else:
            return 'Outro'
    except:
        return 'Outro'

# Função para contar as ocorrências de cada trimestre e calcular as porcentagens
def contar_resposta_unica_trimestre(df):
    df['Trimestre'] = df.iloc[:, 9].apply(classificar_trimestre)
    contagem_trimestre = df['Trimestre'].value_counts()
    porcentagens_trimestre = (contagem_trimestre / contagem_trimestre.sum()) * 100
    porcentagens_trimestre = porcentagens_trimestre.round(2)
    trimestre_df = pd.DataFrame({'Trimestre': contagem_trimestre.index, 'Contagem': contagem_trimestre.values, 'Porcentagem': porcentagens_trimestre.values})
    print(trimestre_df)
    print("\n")
    return trimestre_df

### Análise de Idade

In [198]:
# Calculando os parâmetros de idade

# Calcular a média da coluna "Idade"
media_idade = df['Idade (anos)'].mean().round(2)

# Calcular a mediana da coluna "Idade"
mediana_idade = df['Idade (anos)'].median()

# Calcular o desvio padrão da coluna "Idade"
desvio_padrao_idade = df['Idade (anos)'].std().round(2)

# Calcular o máximo da coluna "Idade"
maximo_idade = df['Idade (anos)'].max()

# Calcular o mínimo da coluna "Idade"
minimo_idade = df['Idade (anos)'].min()

# Apresentar os resultados
print("Média da Idade:", media_idade)
print("Mediana da Idade:", mediana_idade)
print("Desvio Padrão da Idade:", desvio_padrao_idade)
print("Máximo da Idade:", maximo_idade)
print("Mínimo da Idade:", minimo_idade)

# Contagem de consultas pré-natal maior que 6
maior_igual_que_mediana = (df['Idade (anos)'] >= mediana_idade).sum()

# Contagem de consultas pré-natal menor ou igual a 6
menor_que_mediana = (df['Idade (anos)'] < mediana_idade).sum()

print("Idade maior ou igual a mediana:", maior_igual_que_mediana)
print("Idade menor que mediana:", menor_que_mediana)

Média da Idade: 26.85
Mediana da Idade: 27.0
Desvio Padrão da Idade: 5.31
Máximo da Idade: 39
Mínimo da Idade: 15
Idade maior ou igual a mediana: 21
Idade menor que mediana: 18


### Análise de Dados 1

In [199]:
import pandas as pd

estado_civil_df = contar_resposta_unica('Estado civil', df)

prenatal_parceiro_df = contar_resposta_unica('Se tiver companheiro, ele faz o pré-natal do parceiro?', df)

local_pintura_df = contar_resposta_unica('Local onde a pintura foi realizada:', df)

sexo_bebe_df = contar_resposta_unica('Sexo do bebê da arte', df)

                                        Estado civil  Contagem  Porcentagem
0  Convive com companheiro (casada, união estável..)        31        79.49
1               Sem companheiro (solteira, separada)         8        20.51


  Se tiver companheiro, ele faz o pré-natal do parceiro?  Contagem  \
0                                                Não            22   
1                                                Sim            15   

   Porcentagem  
0        59.46  
1        40.54  


     Local onde a pintura foi realizada:  Contagem  Porcentagem
0            Unidade de Saúde Tito Silva        20        51.28
1                                   UFPB         9        23.08
2                                Em casa         7        17.95
3  Unidade de Saúde Mangabeira Integrada         3         7.69


  Sexo do bebê da arte  Contagem  Porcentagem
0               Menino        23        58.97
1               Menina        15        38.46
2             Não sabe         1         2.56


In [200]:
trimestre_df = contar_resposta_unica_trimestre(df)

risco = contar_resposta_unica('Classificação do Risco da gestação', df)

consultas_prenatal = contar_resposta_unica('Número de consultas pré-natal:', df)

            Trimestre  Contagem  Porcentagem
0  Terceiro Trimestre        35        89.74
1   Segundo Trimestre         4        10.26


  Classificação do Risco da gestação  Contagem  Porcentagem
0                     Risco habitual        32        82.05
1                         Alto risco         7        17.95


    Número de consultas pré-natal:  Contagem  Porcentagem
0                                4         7        17.95
1                                7         5        12.82
2                               10         5        12.82
3                                6         4        10.26
4                                9         4        10.26
5                                8         3         7.69
6                                5         3         7.69
7                                3         3         7.69
8                               11         2         5.13
9                               12         1         2.56
10                              23         

In [201]:
media_consultas = round(df['Número de consultas pré-natal:'].mean(),2)

# Contagem de consultas pré-natal maior que 6
maior_igual_que_6 = (df['Número de consultas pré-natal:'] >= 6).sum()

# Contagem de consultas pré-natal menor ou igual a 6
menor_que_6 = (df['Número de consultas pré-natal:'] < 6).sum()

# print("Média Consultas pré-natal maior que 6:", media_consultas)
# print("Consultas pré-natal maior ou igual a 6:", maior_igual_que_6)
# print("Consultas pré-natal menor que 6:", menor_que_6)

# Criar um novo DataFrame com os resultados
consultas_prenatal = pd.DataFrame({
    'Descrição': [
        'Média Consultas pré-natal maior que 6',
        'Consultas pré-natal maior ou igual a 6',
        'Consultas pré-natal menor que 6'
    ],
    'Contagem': [
        media_consultas,
        maior_igual_que_6,
        menor_que_6
    ],
    'Porcentagem': [
        '',  # Para a média, não há porcentagem
        round(maior_igual_que_6, 2),
        round(menor_que_6, 2)
    ]
})



In [202]:
gest_planejada = contar_resposta_unica('Gestação planejada?', df)

gest_desejada = contar_resposta_unica('Gestação desejada?', df)

  Gestação planejada?  Contagem  Porcentagem
0                 Não        30        76.92
1                 Sim         9        23.08


  Gestação desejada?  Contagem  Porcentagem
0                Sim        37        94.87
1                Não         2         5.13




In [203]:
num_gest = contar_resposta_unica('Paridade_ número de gestações (GESTA)', df)

num_partos = contar_resposta_unica('Paridade_ número de PARTOS ', df)

num_abortos = contar_resposta_unica('Número de abortos (abortamentos)', df)


   Paridade_ número de gestações (GESTA)  Contagem  Porcentagem
0                                      1        21        53.85
1                                      2        12        30.77
2                                      3         3         7.69
3                                      4         2         5.13
4                                      7         1         2.56


   Paridade_ número de PARTOS   Contagem  Porcentagem
0                            0        18        46.15
1                            1        12        30.77
2                            2         6        15.38
3                            3         2         5.13
4                            4         1         2.56


   Número de abortos (abortamentos)  Contagem  Porcentagem
0                                 0        33        84.62
1                                 1         5        12.82
2                                 2         1         2.56




In [204]:
sabe_oque_e = contar_resposta_unica('Você sabe o que é Arte Gestacional?', df)

conheceu_como = contar_resposta_unica('Como conheceu a Arte Gestacional?', df)

  Você sabe o que é Arte Gestacional?  Contagem  Porcentagem
0                                 Sim        28        71.79
1                                 Não        11        28.21


  Como conheceu a Arte Gestacional?  Contagem  Porcentagem
0         Projeto Aconchego Materno        17        43.59
1                          Internet        12        30.77
2                             Doula         4        10.26
3                            Amigos         2         5.13
4               Grupo das Gestantes         1         2.56
5                   Evento de Saúde         1         2.56
6                         Faculdade         1         2.56
7                      Não conhecia         1         2.56




In [205]:
acompanhante = contar_resposta_unica('Está com acompanhante durante a arte gestacional?', df)

arte_anteriormente = contar_resposta_unica('Já fez a arte gestacional anteriormente?', df)

plano_parto = contar_resposta_unica('Fez plano de parto?', df)

  Está com acompanhante durante a arte gestacional?  Contagem  Porcentagem
0                                               Não        17        43.59
1                           Sim, não informado quem        11        28.21
2                                     Sim, parceiro         5        12.82
3                                          Sim, mãe         1         2.56
4                                     Sim, filho(a)         1         2.56
5                Gostaria que o filho tivesse vindo         1         2.56
6                                  Sim, mãe e irmão         1         2.56
7                                          Filho(a)         1         2.56
8                                      Sim, família         1         2.56


  Já fez a arte gestacional anteriormente?  Contagem  Porcentagem
0                                      Não        38        97.44
1                                      Sim         1         2.56


  Fez plano de parto?  Contagem  Porcentagem
0  

In [206]:
atividades_educativas = contar_resposta_unica('Participou de atividades educativas?', df)

visita_maternidade = contar_resposta_unica('Realizou visita a maternidade?', df)

  Participou de atividades educativas?  Contagem  Porcentagem
0                                  Sim        22        56.41
1                                  Não        17        43.59


  Realizou visita a maternidade?  Contagem  Porcentagem
0                            Não        21        53.85
1                            Sim        18        46.15




In [207]:
emocao_durante = contar_multiplas_respostas('Qual emoção desencadeada durante a pintura?', df)

emocao_apos = contar_multiplas_respostas('Qual emoção desencadeada após a pintura?', df)


  Qual emoção desencadeada durante a pintura?  Contagem  Porcentagem
0                                     Alegria        38        97.44
1                                        Amor         1         2.56
2                                   Ansiedade         1         2.56
3                                     Cuidado         1         2.56
4                                  Felicidade         2         5.13
5                Preocupada com a outra filha         1         2.56
6                                  Satisfação         1         2.56
7                                    Surpresa        11        28.21
8                               Tranquilidade         1         2.56


  Qual emoção desencadeada após a pintura?  Contagem  Porcentagem
0                                  Alegria        36        92.31
1                                     Amor         2         5.13
2                                 Gratidão         2         5.13
3                                 Perfeito  

In [208]:
consciencia_fetal = contar_resposta_unica('A arte te ajudou a ter consciência fetal (posição, parto)?', df)

curiosidade_info = contar_resposta_unica('A arte te despertou curiosidade para buscar mais informações sobre o parto?', df)

proxima_do_bebe = contar_resposta_unica('Você se sentiu mais próxima do seu bebê com a arte?', df)

  A arte te ajudou a ter consciência fetal (posição, parto)?  Contagem  \
0                                                Sim                37   
1                                                Não                 2   

   Porcentagem  
0        94.87  
1         5.13  


  A arte te despertou curiosidade para buscar mais informações sobre o parto?  \
0                                                Sim                            
1                                                Não                            

   Contagem  Porcentagem  
0        30        76.92  
1         9        23.08  


  Você se sentiu mais próxima do seu bebê com a arte?  Contagem  Porcentagem
0                                                Sim         39        100.0




In [209]:
comportamentos_maternos = contar_multiplas_respostas('Comportamentos maternos gerados pela arte gestacional:', df)

   Comportamentos maternos gerados pela arte gestacional:  Contagem  \
0            Busca por informações e proteção do bebê            13   
1                   Busca por interação com a família            15   
2                    Busca por proximidade com o bebê            22   
3   Buscar conhecer sobre o bebê e seu estado intr...            14   
4           Consciência em relação as posições fetais            22   
5                                     Imaginar o bebê            34   
6                                              Prazer            24   
7                    Proximidade e conexão com o bebê            29   
8                                         Relaxamento            26   
9                    Sentir-se confiante para o parto            21   
10                                    Sentir-se feliz            32   
11            Sentir-se triste com a perda da pintura            12   
12                     Valorização pessoa da gestante            30   

    P

In [210]:
cores = contar_multiplas_respostas('Cores escolhidas para a arte gestacional', df)

oferta = contar_multiplas_respostas('Durante a pintura houve oferta de:', df)

  Cores escolhidas para a arte gestacional  Contagem  Porcentagem
0                                  Amarelo         9        23.08
1                         Azul e seus tons        13        33.33
2                                   Branco        10        25.64
3                                  Laranja         6        15.38
4                                   Marron         1         2.56
5                         Rosa e seus tons        10        25.64
6               Roxo (violeta) e seus tons         6        15.38
7                        Verde e seus tons        13        33.33
8                     Vermelho e seus tons         5        12.82


  Durante a pintura houve oferta de:  Contagem  Porcentagem
0                       Aromaterapia         8        20.51
1                        Escalda pés        13        33.33
2  Música de preferência da gestante        33        84.62
3     Ritual de despedida da barriga         9        23.08




In [211]:
repetir_arte = contar_resposta_unica('Realizaria a arte gestacional novamente?', df)

indicaria_arte = contar_resposta_unica('Indicaria a arte gestacional para conhecidas?', df)

  Realizaria a arte gestacional novamente?  Contagem  Porcentagem
0                                      Sim        37        94.87
1                                      Não         2         5.13


  Indicaria a arte gestacional para conhecidas?  Contagem  Porcentagem
0                                           Sim        39        100.0




In [212]:
uma_palavra = contar_multiplas_respostas('Represente em uma palavra como se sentiu com a pintura:', df)

# uma_palavra.to_excel('uma_palavra.xlsx')


   Represente em uma palavra como se sentiu com a pintura:  Contagem  \
0                                               Feliz             15   
1                                             Ansiosa              2   
2                                             Cuidado              2   
3                                          Emocionada              2   
4                                          Felicidade              2   
5                                           Confiança              1   
6                                        Entusiasmada              1   
7                                           Tranquila              1   
8                                           Realizada              1   
9                                        Significante              1   
10                                              Amada              1   
11                                        Maravilhosa              1   
12                                             Pronta           

In [213]:
desenhos = contar_multiplas_respostas('Descreva a ilustração presente na arte gestacional', df)

# desenhos.to_excel('desenhos.xlsx')


   Descreva a ilustração presente na arte gestacional  Contagem  Porcentagem
0                                             Animais         1         2.56
1                                           Arco-íris         1         2.56
2                                               Balão         1         2.56
3                                               Barco         1         2.56
4                                                Bebê         4        10.26
5                                     Bola de futebol         2         5.13
6                                          Borboletas         3         7.69
7                                              Brinco         2         5.13
8                            Carinhas que a filha fez         1         2.56
9                                    Cordão umbilical         2         5.13
10                                              Coroa         1         2.56
11                                                Céu         2         5.13

In [214]:
def contar_cores_por_sexo(coluna, df):
    # Separar as cores em colunas diferentes
    cores_sep = df[coluna].str.split(', ', expand=True)
    
    # Adicionar a coluna 'Sexo do bebê da arte' para o DataFrame resultante
    cores_sep['Sexo do bebê da arte'] = df['Sexo do bebê da arte']
    
    # Transformar as colunas de cores em uma única coluna e colocar tudo em uma única linha
    cores_melted = pd.melt(cores_sep, id_vars=['Sexo do bebê da arte'], value_name='Cor')
    
    # Remover linhas com valores nulos na coluna 'Cor'
    cores_melted.dropna(subset=['Cor'], inplace=True)
    
    # Contar o número de ocorrências de cada cor para cada sexo
    contagem_cores = cores_melted.groupby(['Sexo do bebê da arte', 'Cor']).size().reset_index(name='Contagem')
    
    # Pivotar a tabela para ter uma única linha para cada cor, com as contagens para cada sexo
    contagem_cores_pivot = contagem_cores.pivot(index='Cor', columns='Sexo do bebê da arte', values='Contagem').fillna(0).astype(int)
    
    # Adicionar a coluna com a soma das contagens para todos os sexos
    contagem_cores_pivot['Total'] = contagem_cores_pivot.sum(axis=1)
    
    # Adicionar a porcentagem de cada sexo em relação ao total
    contagem_cores_pivot['Porcentagem (Meninas)'] = (contagem_cores_pivot['Menina'] / 15 * 100).round(2)
    contagem_cores_pivot['Porcentagem (Meninos)'] = (contagem_cores_pivot['Menino'] / 23 * 100).round(2)
    contagem_cores_pivot['Porcentagem (Desconhecido)'] = (contagem_cores_pivot['Não sabe'] / 1 * 100).round(2)

    return contagem_cores_pivot

# Exemplo de uso da função para contar as cores escolhidas para a arte gestacional por sexo do bebê
cores_por_sexo = contar_cores_por_sexo('Cores escolhidas para a arte gestacional', df)
# cores_por_sexo.to_excel('Cores_Por_Sexo.xlsx')


## A partir daqui, começa a seguinte análise:

Número de consultas com as variáveis:
    A arte despertando consciência das posições fetais (arte_consciencia_fetal)
    A arte despertando curiosidade para buscar mais informações sobre o parto (arte_curiosidade_parto)
    Comportamentos positivos (comportamentos_positivos)

Gestação desejada/não desejada com:
    Sentimentos (sentimentos)
    Você se sentiu mais próxima do seu bebê com a arte? (proximidade_bebe)

In [215]:
import pandas as pd
import numpy as np
from scipy.stats import ttest_ind, chi2_contingency
import statsmodels.api as sm
import seaborn as sns
import matplotlib.pyplot as plt

# Carregar os dados
file_path = 'Questionário Artes Gestacional (respostas).xlsx'
data = pd.read_excel(file_path)

# Função para definir se possui companheiro
def possui_companheiro(estado_civil):
    if 'Convive com companheiro' in estado_civil:
        return 'Possui Companheiro'
    else:
        return 'Não Possui Companheiro'

# Aplicar a função para criar uma nova coluna
data['possui_companheiro'] = data['Estado civil'].apply(possui_companheiro)

# Renomear colunas para facilitar a manipulação
data = data.rename(columns={
    'Número de consultas pré-natal:': 'num_consultas',
    'A arte te ajudou a ter consciência fetal (posição, parto)?': 'arte_consciencia_fetal',
    'A arte te despertou curiosidade para buscar mais informações sobre o parto?': 'arte_curiosidade_parto',
    'Comportamentos maternos gerados pela arte gestacional:': 'comportamentos_positivos',
    'Gestação desejada?': 'gestacao_desejada',
    'Qual emoção desencadeada durante a pintura?': 'sentimentos',
    'Você se sentiu mais próxima do seu bebê com a arte?': 'proximidade_bebe',
    'possui_companheiro': 'estado_civil'  # Adicionado para renomear a coluna de estado civil
})

# Converter variáveis categóricas em numéricas
data['arte_consciencia_fetal'] = data['arte_consciencia_fetal'].map({'Sim': 1, 'Não': 0})
data['arte_curiosidade_parto'] = data['arte_curiosidade_parto'].map({'Sim': 1, 'Não': 0})
data['comportamentos_positivos'] = data['comportamentos_positivos'].apply(lambda x: 1 if x else 0)
data['gestacao_desejada'] = data['gestacao_desejada'].map({'Sim': 1, 'Não': 0})
data['proximidade_bebe'] = data['proximidade_bebe'].map({'Sim': 1, 'Não': 0})

# Codificar manualmente a coluna estado_civil
data['estado_civil_encoded'] = data['estado_civil'].astype('category').cat.codes

# Estatísticas descritivas
print(data.describe())



       Idade (anos)  num_consultas  gestacao_desejada  \
count     39.000000      39.000000          39.000000   
mean      26.846154       7.512821           0.948718   
std        5.309071       3.831118           0.223456   
min       15.000000       3.000000           0.000000   
25%       23.500000       4.500000           1.000000   
50%       27.000000       7.000000           1.000000   
75%       31.000000       9.500000           1.000000   
max       39.000000      23.000000           1.000000   

       Paridade_ número de gestações (GESTA)  Paridade_ número de PARTOS   \
count                              39.000000                    39.000000   
mean                                1.769231                     0.871795   
std                                 1.202225                     1.030572   
min                                 1.000000                     0.000000   
25%                                 1.000000                     0.000000   
50%                     

In [216]:


# Correlações de Pearson (convertendo apenas as colunas numéricas)
numeric_cols = ['num_consultas', 'arte_consciencia_fetal', 'arte_curiosidade_parto', 'comportamentos_positivos', 'gestacao_desejada', 'proximidade_bebe', 'estado_civil_encoded']
pearson_corr = data[numeric_cols].corr(method='pearson')
print(pearson_corr)

# Correlações de Spearman
spearman_corr = data[numeric_cols].corr(method='spearman')
print(spearman_corr)

# Teste t de Student
gestacao_sim = data[data['gestacao_desejada'] == 1]['num_consultas']
gestacao_nao = data[data['gestacao_desejada'] == 0]['num_consultas']
t_stat, p_val = ttest_ind(gestacao_sim, gestacao_nao)
print('T-statistic:', t_stat)
print('P-value:', p_val)

# Regressão Logística
X = data[['num_consultas', 'arte_consciencia_fetal', 'arte_curiosidade_parto']]
y = data['gestacao_desejada']
X = sm.add_constant(X)
log_reg = sm.Logit(y, X).fit()
print(log_reg.summary())

# Chi-quadrado de Independência
contingency_table = pd.crosstab(data['estado_civil_encoded'], data['gestacao_desejada'])
chi2, p, dof, ex = chi2_contingency(contingency_table)
print('Chi-square:', chi2)
print('P-value:', p)


                          num_consultas  arte_consciencia_fetal  \
num_consultas                  1.000000               -0.152910   
arte_consciencia_fetal        -0.152910                1.000000   
arte_curiosidade_parto         0.170832                0.424476   
comportamentos_positivos            NaN                     NaN   
gestacao_desejada             -0.368088               -0.054054   
proximidade_bebe                    NaN                     NaN   
estado_civil_encoded           0.152846               -0.118108   

                          arte_curiosidade_parto  comportamentos_positivos  \
num_consultas                           0.170832                       NaN   
arte_consciencia_fetal                  0.424476                       NaN   
arte_curiosidade_parto                  1.000000                       NaN   
comportamentos_positivos                     NaN                       NaN   
gestacao_desejada                      -0.127343                       Na

         Current function value: 0.151820
         Iterations: 35
                           Logit Regression Results                           
Dep. Variable:      gestacao_desejada   No. Observations:                   39
Model:                          Logit   Df Residuals:                       35
Method:                           MLE   Df Model:                            3
Date:                Tue, 25 Jun 2024   Pseudo R-squ.:                  0.2494
Time:                        22:20:20   Log-Likelihood:                -5.9210
converged:                      False   LL-Null:                       -7.8886
Covariance Type:            nonrobust   LLR p-value:                    0.2685
                             coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                     24.2548   4.04e+04      0.001      1.000   -7.91e+04    7.91e+04
num_consultas             -0.

/home/ruanv/.local/lib/python3.10/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


Vamos analisar os resultados das correlações e da regressão logística com mais detalhes:

### Correlações de Pearson
As correlações de Pearson nos dão uma ideia da relação linear entre as variáveis contínuas e binárias. Aqui estão algumas das correlações observadas:

1. **Número de consultas pré-natal vs. Arte despertando consciência fetal**:
   - Correlação: -0.15 (correlação negativa fraca)
   - Interpretação: Há uma leve tendência de que um maior número de consultas esteja associado a uma menor percepção de que a arte ajuda na consciência fetal.

2. **Número de consultas pré-natal vs. Arte despertando curiosidade sobre o parto**:
   - Correlação: 0.17 (correlação positiva fraca)
   - Interpretação: Há uma leve tendência de que um maior número de consultas esteja associado a uma maior curiosidade sobre o parto despertada pela arte.

3. **Número de consultas pré-natal vs. Gestação desejada**:
   - Correlação: -0.37 (correlação negativa moderada)
   - Interpretação: Há uma tendência moderada de que um maior número de consultas esteja associado a gestações não desejadas.

4. **Estado civil codificado vs. Arte despertando curiosidade sobre o parto**:
   - Correlação: -0.48 (correlação negativa moderada)
   - Interpretação: Participantes que não são casadas têm maior probabilidade de achar que a arte desperta curiosidade sobre o parto.

### Correlações de Spearman
As correlações de Spearman são utilizadas para variáveis ordinais e categóricas. No seu caso, algumas correlações não puderam ser calculadas devido à falta de variabilidade em algumas variáveis, como comportamentos positivos e proximidade com o bebê.

### Regressão Logística
A regressão logística foi realizada para modelar a variável binária `gestacao_desejada` (0 = não desejada, 1 = desejada) com as variáveis preditoras `num_consultas`, `arte_consciencia_fetal` e `arte_curiosidade_parto`.

- **Coeficientes e Significância**:
  - `num_consultas`: Coeficiente de -0.2324, p-valor de 0.097. Sugere que um maior número de consultas pode estar levemente associado a uma menor probabilidade de a gestação ser desejada, mas o efeito não é estatisticamente significativo ao nível de 5%.
  - `arte_consciencia_fetal`: Coeficiente de -0.4420, p-valor de 1.000. Não há evidência significativa de que a arte despertando consciência fetal esteja associada à gestação desejada.
  - `arte_curiosidade_parto`: Coeficiente de -18.9078, p-valor de 0.999. Não há evidência significativa de que a arte despertando curiosidade sobre o parto esteja associada à gestação desejada.

- **Pseudo R-squared**: 0.2494. Indica que aproximadamente 24.94% da variabilidade na gestação desejada é explicada pelas variáveis preditoras no modelo.

- **Convergência**: O modelo não convergiu corretamente, indicando possíveis problemas como separação perfeita (quasi-separation), o que significa que algumas observações podem ser perfeitamente preditas pelas variáveis preditoras. Isso pode ocorrer devido ao pequeno tamanho da amostra ou à falta de variabilidade nas respostas.

### Interpretação Geral
1. **Relações Fracas**: A maioria das correlações observadas é fraca, sugerindo que não há relações lineares fortes entre as variáveis analisadas.
2. **Número de Consultas e Gestação Desejada**: Há uma tendência moderada de que mais consultas pré-natal estejam associadas a gestações não desejadas, mas essa relação não é forte o suficiente para ser conclusiva.
3. **Problemas na Regressão Logística**: A regressão logística teve problemas de convergência e não encontrou relações significativas entre as variáveis preditoras e a gestação desejada, possivelmente devido ao tamanho da amostra ou à distribuição dos dados.

Esses resultados indicam que pode ser necessário revisar a coleta de dados ou aumentar o tamanho da amostra para obter resultados mais robustos. Além disso, explorar outras técnicas estatísticas ou incluir outras variáveis preditoras pode ajudar a entender melhor as relações entre as variáveis do seu estudo.

Salvando os dataframes em um excel só:

In [217]:
# Lista de DataFrames


dfs = [estado_civil_df, prenatal_parceiro_df, local_pintura_df, sexo_bebe_df, trimestre_df, risco, consultas_prenatal, gest_planejada, gest_desejada, num_gest, num_partos, num_abortos, sabe_oque_e, conheceu_como, acompanhante, arte_anteriormente, plano_parto, atividades_educativas, visita_maternidade, emocao_durante, emocao_apos, consciencia_fetal, curiosidade_info, proxima_do_bebe, comportamentos_maternos, cores, oferta, repetir_arte, indicaria_arte, uma_palavra, desenhos, cores_por_sexo]

# Criar um objeto ExcelWriter do Pandas
writer = pd.ExcelWriter('Tabelas de Saída.xlsx', engine='xlsxwriter')

# Iterar sobre os DataFrames e escrever cada um em uma planilha
for i, df in enumerate(dfs, start=1):
    sheet_name = f'DF_{i}'
    df.to_excel(writer, sheet_name=sheet_name, index=False)

# Salvar o arquivo Excel
writer.close()